core/services/keeper/upkeep_executer.go

```diff
@@ -66,135 +70,131 @@ func NewUpkeepExecuter(
+	return ex.StartOnce("UpkeepExecuter", func() error {
+		ex.wgDone.Add(2)
+		go ex.run()
+		latestHead, unsubscribeHeads := ex.headBroadcaster.Subscribe(ex)
 		if latestHead != nil {
-			executer.mailbox.Deliver(*latestHead)
+			ex.mailbox.Deliver(*latestHead)
 		}
 		go func() {
 			defer unsubscribeHeads()
-			defer executer.wgDone.Done()
-			<-executer.chStop
+			defer ex.wgDone.Done()
+			<-ex.chStop
 		}()
 		return nil
 	})
 }
 
-func (executer *UpkeepExecuter) Close() error {
-	return executer.StopOnce("UpkeepExecuter", func() error {
-		close(executer.chStop)
-		executer.wgDone.Wait()
+// Close stops and closes upkeep executer
+func (ex *UpkeepExecuter) Close() error {
+	return ex.StopOnce("UpkeepExecuter", func() error {
+		close(ex.chStop)
+		ex.wgDone.Wait()
 		return nil
 	})
 }
 
-func (executer *UpkeepExecuter) OnNewLongestChain(ctx context.Context, head models.Head) {
-	executer.mailbox.Deliver(head)
+// OnNewLongestChain handles the given head of a new longest chain
+func (ex *UpkeepExecuter) OnNewLongestChain(ctx context.Context, head models.Head) {
+	ex.mailbox.Deliver(head)
 }
 
-func (executer *UpkeepExecuter) run() {
-	defer executer.wgDone.Done()
+func (ex *UpkeepExecuter) run() {
+	defer ex.wgDone.Done()
 	for {
 		select {
-		case <-executer.chStop:
+		case <-ex.chStop:
 			return
-		case <-executer.mailbox.Notify():
-			executer.processActiveUpkeeps()
+		case <-ex.mailbox.Notify():
+			ex.processActiveUpkeeps()
 		}
 	}
 }
 
-func (executer *UpkeepExecuter) processActiveUpkeeps() {
+func (ex *UpkeepExecuter) processActiveUpkeeps() {
 	// Keepers could miss their turn in the turn taking algo if they are too overloaded
 	// with work because processActiveUpkeeps() blocks
-	item, exists := executer.mailbox.Retrieve()
+	item, exists := ex.mailbox.Retrieve()
 	if !exists {
-		logger.Info("UpkeepExecuter: no head to retrieve. It might have been skipped")
+		ex.logger.Info("UpkeepExecuter: no head to retrieve. It might have been skipped")
 		return
 	}
 
 	head, ok := item.(models.Head)
 	if !ok {
-		logger.Errorf("expected `models.Head`, got %T", head)
+		ex.logger.Errorf("UpkeepExecuter: expected `models.Head`, got %T", head)
 		return
 	}
 
-	logger.Debugw("UpkeepExecuter: checking active upkeeps", "blockheight", head.Number, "jobID", executer.job.ID)
+	ex.logger.Debugw("UpkeepExecuter: checking active upkeeps", "blockheight", head.Number)
 
 	ctx, cancel := postgres.DefaultQueryCtx()
 	defer cancel()
 
-	activeUpkeeps, err := executer.orm.EligibleUpkeepsForRegistry(
+	activeUpkeeps, err := ex.orm.EligibleUpkeepsForRegistry(
 		ctx,
-		executer.job.KeeperSpec.ContractAddress,
+		ex.job.KeeperSpec.ContractAddress,
 		head.Number,
-		executer.config.KeeperMaximumGracePeriod(),
+		ex.config.KeeperMaximumGracePeriod(),
 	)
 	if err != nil {
-		logger.Errorf("unable to load active registrations: %v", err)
+		ex.logger.WithError(err).Error("UpkeepExecuter: unable to load active registrations")
```

Comment:
Same thing - can we remove the `"UpkeepExecuter:` portion of the error message?

Reply:
Done